build/source/engine/computFlux.f90

```diff
@@ -0,0 +1,851 @@
+ ! input: diagnostic variables
+ real(dp),intent(in)             :: scalarCanopyLiqTrial      ! trial value for mass of liquid water on the vegetation canopy (kg m-2)
+ real(dp),intent(in)             :: scalarCanopyIceTrial      ! trial value for mass of ice on the vegetation canopy (kg m-2)
+ real(dp),intent(in)             :: mLayerVolFracLiqTrial(:)  ! trial value for volumetric fraction of liquid water (-)
+ real(dp),intent(in)             :: mLayerVolFracIceTrial(:)  ! trial value for volumetric fraction of ice (-)
+ ! input: data structures
+ type(model_options),intent(in)  :: model_decisions(:)        ! model decisions
+ type(var_i),        intent(in)  :: type_data                 ! type of vegetation and soil
+ type(var_d),        intent(in)  :: attr_data                 ! spatial attributes
+ type(var_d),        intent(in)  :: mpar_data                 ! model parameters
+ type(var_d),        intent(in)  :: forc_data                 ! model forcing data
+ type(var_dlength),  intent(in)  :: bvar_data                 ! model variables for the local basin
+ type(var_dlength),  intent(in)  :: prog_data                 ! prognostic variables for a local HRU
+ type(var_ilength),  intent(in)  :: indx_data                 ! indices defining model states and layers
+ ! input-output: data structures
+ type(var_dlength),intent(inout) :: diag_data                 ! diagnostic variables for a local HRU
+ type(var_dlength),intent(inout) :: flux_data                 ! model fluxes for a local HRU
+ type(var_dlength),intent(inout) :: deriv_data                ! derivatives in model fluxes w.r.t. relevant state variables
+ ! input-output: flux vector and baseflow derivatives
+ integer(i4b),intent(inout)      :: ixSaturation              ! index of the lowest saturated layer (NOTE: only computed on the first iteration)
+ real(dp),intent(out)            :: dBaseflow_dMatric(:,:)    ! derivative in baseflow w.r.t. matric head (s-1)
+ real(dp),intent(out)            :: fluxVec(:)                ! model flux vector (mixed units)
+ ! output: error control
+ integer(i4b),intent(out)        :: err                       ! error code
+ character(*),intent(out)        :: message                   ! error message
+ ! ---------------------------------------------------------------------------------------
+ ! * local variables
+ ! ---------------------------------------------------------------------------------------
+ integer(i4b)                    :: local_ixGroundwater       ! local index for groundwater representation
+ integer(i4b)                    :: iSoil                     ! index of soil layer
+ integer(i4b)                    :: iLayer                    ! index of model layers
+ real(dp)                        :: theta                     ! liquid water equivalent of total water (liquid plus ice)
+ real(dp),parameter              :: canopyTempMax=500._dp     ! expected maximum value for the canopy temperature (K)
```

Comment:
True. Physical constants do belong in an external module.

Suggest we think of the different types of global constants. These can include:
* Physical constants
* Limits used for error checking (as is done here)
* Flags (e.g., missing values)

Once we have the different categories defined we can reorganize the code to clarify the role of the different constants (and avoid the circuitous problems identified in the comment above). 

This reorganization should be the topic of a future PR, and I've opened up an issue to address this (issue #140).